src/Workspaces/Remote/ServiceHub/Services/CodeLensReferences/RemoteCodeLensReferencesService.cs

```diff
@@ -43,18 +43,21 @@ public RemoteCodeLensReferencesService(in ServiceConstructionArguments arguments
             {
                 using (Logger.LogBlock(FunctionId.CodeAnalysisService_GetReferenceCountAsync, documentId.ProjectId.DebugName, cancellationToken))
                 {
-                    var solution = await GetSolutionAsync(solutionInfo, cancellationToken).ConfigureAwait(false);
-                    var syntaxNode = await TryFindNodeAsync(solution, documentId, textSpan, cancellationToken).ConfigureAwait(false);
-                    if (syntaxNode == null)
-                    {
-                        return null;
-                    }
+                    return await RunWithSolutionAsync(solutionInfo, async solution =>
```

Comment:
Can't we use RunServiceWithSolutionAsync?

Reply:
the issue here is preserving the meaning of the using-block for telemetry.

Reply:
could LogBlock be around RunServiceAsync?